### ISSUE TITLE
Document operators defined for concepts.

### ISSUE THREAD

**Issue by ricejasonf** [boostorg/hana#213]
In `include/boost/hana/detail/operators/*.hpp` there are operators defined for concepts that aren't making it into the documentation. They don't show up in the search bar, and I was unable to find them in side-menu including `Details -> adl`.

One example is the `operator|` for the Monad `chain` function.

**Comment by ldionne** [comment]
Actually, these operators are defined per-datatype. Indeed, they are not provided by the concept, but by the container. For example, see [hana::tuple's documentation](http://boostorg.github.io/hana/structboost_1_1hana_1_1tuple.html). There's a section called _provided operators_.

**Comment by ricejasonf** [comment]
Aha. I noticed some operators show up in the side menu and the search bar but not others.

**Comment by ldionne** [comment]
Which ones?

You know what; I think we can do better by actually documenting `operator|` & friends with Doxygen for each container. Something like

```c++
#ifdef BOOST_HANA_DOXYGEN_INVOKED
//! Equivalent to `hana::chain`.
template <typename ...T, typename F>
constexpr auto operator|(tuple<T...>, F);
#endif
```

**Comment by ldionne** [comment]
This is fixed by PR #250.
